api.go

```diff
@@ -33,37 +33,20 @@ func (a *API) HandleDirectory(dir string) error {
 		return fmt.Errorf("Cannot list clang-c directory: %v", err)
 	}
 
-	h := &HeaderFile{
-		api: a,
-
-		dir: dir,
-
-		lookupEnum:        map[string]*Enum{},
-		lookupNonTypedefs: map[string]string{},
-		lookupStruct: map[string]*Struct{
-			"cxstring": &Struct{
-				Name:  "cxstring",
-				CName: "CXString",
-			},
-		},
-	}
-
 	for _, hf := range headers {
-		name := hf.Name()
-
-		if hf.IsDir() || !strings.HasSuffix(name, ".h") {
+		if hf.IsDir() || !strings.HasSuffix(hf.Name(), ".h") {
 			continue
 		}
-
-		h.name = dir + name
+        
+        h := newHeaderFile(a, dir + hf.Name(), dir)
```

Comment:
"Dir" is used later on, see headerfile:173 (updated version) 
If you ask why dir was in the headerfile name (name = dir + hf.Name()) in the first place, you would have to ask yourself ;) I just reused the same logic so to speak, but found it always peculiar. 
But it has a slight convenience for error messages, to get full path immediately